pallas-primitives/src/conway/model.rs

```diff
@@ -27,25 +27,216 @@ pub use crate::babbage::OperationalCert;
 
 pub use crate::babbage::Header;
 
-pub type Multiasset<A> = BTreeMap<PolicyId, BTreeMap<AssetName, A>>;
+use pallas_codec::minicbor::data::Type;
 
-pub type Mint = Multiasset<NonZeroInt>;
+trait StrictContext {
```

Comment:
Some suggestions:
 - If we name this something more general, like `ValidationContext`,
 - and have the push_error should return a Result that we can `?`;

Then we can have a NoopContext, which does nothing; an AccumulatingContext which accumulates things as errors, and a TerminatingContext, which returns an error.

Then, our `Strict<T>` implements minicbor::Decode, constructs a `TerminatingContext`, and deserializes T;

And we could have a `StrictVerbose<T>` that implements `minicbor::Decode`, constructs an `AccumulatingContext`, and then concatenates all the errors to return it's own at the end.